sql/rpl_trx_boundary_parser.cc

```diff
@@ -110,7 +110,7 @@ Transaction_boundary_parser::get_event_boundary_type(
   if (length < header_size)
     goto end;
 
-  event_type= (Log_event_type)buf[EVENT_TYPE_OFFSET];
+  event_type= (Log_event_type)static_cast<unsigned char>(buf[EVENT_TYPE_OFFSET]);
```

Comment:
Cannot just cast to Log_event_type directly?

Reply:
It cannot be - otherwise buf[EVENT_TYPE_OFFSET] will return signed char - so in case of Start_encryption event - it will be signed number that will be then cast to underlying enum type. We need to first cast it to unsigned char so it would not be cast as <0 number .